docs/examples/01_current_injection.py

```diff
@@ -1,502 +1,694 @@
-# Since it is 1-dimensional, this is equivalent to scaling it by a constant ad adding an intercept
-# $$
-# L(i(t)) = w \cdot i(t) + c
-# $$
-
-w = 0.05
-c = -2
 
-L = w * np.squeeze(input_feature) + c
 
-interval = nap.IntervalSet(start=471, end=471.5)
-plt.figure(figsize=(10, 3.5))
-ax = plt.subplot(1, 2, 1)
-ax.set_title("Current $i(t)$")
-ax.plot(np.squeeze(input_feature.restrict(interval)), color="grey")
-ax.set_xlabel("Time (s)")
-ax = plt.subplot(1, 2, 2)
-ax.set_title("Linearly Transformed $L(t)$")
-ax.plot(np.squeeze(L.restrict(interval)), color="orange")
-ax.set_xlabel("Time (s)")
-plt.tight_layout()
 # %%
-# Then a non-linearity is applied to transform this quantity into a positive firing rate.
-# $$ \lambda (t) = \exp (w \cdot i(t) + c) \tag{1}$$
+#
+# Our last step is to convert these to `jax.numpy` arrays.
 
-# apply an exponential non-linearity to L(t)
-predicted_rate = np.exp(L)
+predictor = jax.numpy.asarray(predictor.values)
+count = jax.numpy.asarray(count.values)
 
 # %%
-# The same linear-non-linear transformation is implemented by the `predict` method of the nemos.glm.GLM
-# object.
+#
+# !!! info
+#
+#     In this example, we're being very explicit about this conversion to
+#     jax.numpy arrays. However, in general, nemos is able to properly convert
+#     from pynapple objects to jax.numpy arrays without any additional steps
+#     (it can similarly convert from numpy arrays to jax.numpy arrays). Thus,
+#     in later tutorials we will omit this step.
+#
+# ### Fitting the model
+#
+# Now we're ready to fit our model!
+#
+# First, we need to define our GLM model object. We intend for our models to be
+# similar to
+# [scikit-learn's](https://scikit-learn.org/stable/getting_started.html)
+# estimators: they are initialized with hyperparameters that specify
+# optimization and model details, and then the user calls `.fit` with the
+# model matrix and data.
+#
+# To initialize our model, we need to specify the regularizer and observation
+# model objects, both of which should be one of our custom objects:
+#
+# - Regularizer: this object specifies both the solver algorithm and the
+#   regularization scheme. They are jointly specified because each
+#   regularization scheme has a list of compatible solvers to choose between.
+#   Regularization modifies the objective function to reflect your prior
+#   beliefs about the parameters, such as sparsity. Regularization becomes more
+#   important as the number of input features, and thus model parameters,
+#   grows. They can be found within `nemos.regularizer`.
+#
+# !!! warning
+#
+#     With a convex problem like the GLM, in theory it does not matter which
+#     solver algorithm you use. In practice, due to numerical issues, it might.
+#     Thus, it's worth trying a couple to see how their solutions compare.
+#     (Different regularization schemes will always give different results.)
```

Comment:
hmm. I wanted to highlight this in case people were confused about why we set the optimizer at all -- when we described the model as convex, one of the selling points was that we'd always find the best solution, so why does it matter.

I think what you're describing sounds like a good idea, but too much to get done before this workshop. I copied your commnets into https://github.com/flatironinstitute/nemos/issues/90 and we'll address it after the workshop.

For this workshop, do you think this caveat is okay?